corehq/sql_db/tests/test_fields.py

```diff
@@ -51,13 +53,18 @@ class Test(models.Model):
         assert has_pattern_ops_index(statements), statements
 
 
+@contextmanager
 def schema_editor():
-    return DatabaseSchemaEditor(connection, collect_sql=True, atomic=False)
+    def capture_sql_output(sql, params=None):
+        editor.collected_sql.append(sql)
+    schema = DatabaseSchemaEditor(connection, collect_sql=True, atomic=False)
+    with patch.object(schema, 'execute', capture_sql_output), schema as editor:
+        yield editor
 
 
 def assert_no_pattern_ops_index(editor):
     statements = editor.collected_sql + editor.deferred_sql
-    assert statements, "expected at least one SQL statement"
+    assert editor.collected_sql, "expected at least one SQL statement"
```

Comment:
I think this does not need to (and should not) change.

Reply:
Sorry. Sloppy on my part. 7d90756008eb0ac9b5fa340f81162ad23b0adde5